contracts/token/ERC721/ERC721.sol

```diff
@@ -120,10 +110,8 @@ contract ERC721 is Context, ERC165, IERC721, IERC721Metadata, IERC721Enumerable
      * @return address currently marked as the owner of the given token ID
      */
     function ownerOf(uint256 tokenId) public view override returns (address) {
-        address owner = _tokenOwner[tokenId];
-        require(owner != address(0), "ERC721: owner query for nonexistent token");
-
-        return owner;
+        require(_tokenOwners.contains(tokenId), "ERC721: owner query for nonexistent token");
+        return _tokenOwners.get(tokenId);
```

Comment:
Can you try to reduce the number of storage reads?

Reply:
Note that we can remove the `require` and the only change would be the revert reason: `get` already performs that same check.

I thought about adding an overload for `get` (and perhaps `at`) that emits a custom error message instead of the generic ones, similar to `SafeMath.sub`. What do you think?

Reply:
Overload sounds good to me.